lib.nix

```diff
@@ -142,15 +143,17 @@ rec
 
       in
         runCommand "dummy-src"
-          { inherit patchedSources cargotomlss; }
+          { inherit copySources copySourcesFrom cargotomlss; }
           ''
             mkdir -p $out/.cargo
             ${lib.optionalString (! isNull cargoconfig) "cp ${config} $out/.cargo/config"}
             cp ${cargolock'} $out/Cargo.lock
 
-            for p in $patchedSources; do
+            for p in $copySources; do
               echo "Copying patched source $p to $out..."
-              cp -R "$p" "$out/"
+              mkdir -p "$out/$p"
+              rmdir "$out/$p"
```

Comment:
Why create the directory then remove it?

Reply:
The trick is `mkdir -p` creates all parent directories, so what this actually does is create all directories up to the parent of `$p`. An alternative would be `mkdir -p "$out/$(dirname "$p")"`, would that look better to you? (it may behave differently for paths with `..`, but these would probably be broken anyway, so…)

Reply:
Ah I see! Either works, in both cases would be great to have a one-liner (`mkdir -p "$out/$p" && rmdir "$out/$p"`) with a comment explaining what's happening. Thanks!